packages/@lwc/engine-core/src/framework/vnodes.ts

```diff
@@ -50,10 +55,24 @@ export interface VScopedSlotFragment extends BaseVNode {
 }
 
 export interface VStaticPart {
+    readonly type: VStaticPartType;
     readonly partId: number;
+    readonly data: VStaticPartData | null;
+    readonly text: string | null;
```

Comment:
Elsewhere in `vnodes.ts` we use `undefined` to indicate a missing value. I know we're inconsistent about this, but I feel like `undefined` is more common to use in these kinds of scenarios than `null`. It's not a big deal though.

Reply:
@nolanlawson in the interest of time, I left it as-is for now but I opened an issue (#4105) for us to do an audit of how we use `null` vs `undefined`.